projectile.el

```diff
@@ -1489,36 +1489,45 @@ Never use on many files since it's going to recalculate the
 project-root for every file."
   (expand-file-name name (projectile-project-root)))
 
-(defun projectile-completing-read (prompt choices &optional initial-input)
+(cl-defun projectile-completing-read (prompt choices
```

Comment:
`cl-defun` allows to have keyword arguments in Elisp, which makes function calls a lot more readable.

Additionally, keyword args allow more flexibility in extending the original interface:

- Adding new keyword args is very easy and breaks nothing.
- Changing keyword args order breaks nothing.
- Removing a keyword arg breaks loudly and is easy to fix.